api/src/main/java/org/regenstrief/linkage/matchresult/DedupMatchResultList.java

```diff
@@ -2,200 +2,92 @@
  */
 package org.regenstrief.linkage.matchresult;
 
-import java.io.File;
-import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.util.ArrayList;
-import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
 import java.util.TreeSet;
 
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-import org.openmrs.module.patientmatching.MatchingConstants;
-import org.openmrs.module.patientmatching.RecordSerializer;
-import org.openmrs.util.OpenmrsUtil;
+import org.openmrs.api.APIException;
+import org.openmrs.module.patientmatching.MatchingStrategy;
+import org.openmrs.module.patientmatching.MatchingStrategyAndStore;
 import org.regenstrief.linkage.MatchResult;
 import org.regenstrief.linkage.Record;
+import org.regenstrief.linkage.util.MatchingConfig;
+import org.regenstrief.linkage.util.ScorePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- *
+ * Probabilistic implementation of a {@link MatchingStrategy}
+ * <p>
+ * TODO Rename to ProbabilisticMatchingStrategy for clarity
+ * </p>
  */
-public class DedupMatchResultList extends MatchResultList {
+public class DedupMatchResultList extends MatchResultList implements MatchingStrategyAndStore {
```

Comment:
I will possibly be renaming this to ProbabilisticMatchingStrategy, am waiting for reviewers to complete not to convolute the changes